### PR TITLE
Add new func to render slice of struct

### PR DESCRIPTION
**Pull request by johejo**

### CODE REVIEW

**Comment by olekukonko** [comment]
@mattn this seems to be useful ....

File: table.go [hunk 1]
```diff
@@ -302,6 +304,69 @@ func (t *Table) SetBorders(border Border) {
...
+	switch vt.Kind() {
+	case reflect.Slice, reflect.Array:
+		if vv.Len() < 1 {
+			return errors.New("empty value")
+		}
+		first := vv.Index(0)
+		e := first.Type()
+		switch e.Kind() {
+		case reflect.Struct:
+			// OK
```
Comment (olekukonko) [inline table.go:323]: what exactly is going to happen here?

Comment (johejo) [reply table.go:323]: Jump to L332 and continue processing.
If I misunderstood the intent of your question, please add a comment.

**Comment by mattn** [comment]
What about pointer fields?

**Comment by johejo** [comment]
Added test cases that support pointer fields.
I'm not an experienced reflect magician, so it might be lacking or wrong with edge cases.

**Comment by johejo** [comment]
Hmm, I found some edge cases...
continue fixing

**Comment by johejo** [comment]
Added more test cases.

**Comment by mattn** [comment]
LGTM

**Comment by mattn** [comment]
Thank you!
